actors/market/src/lib.rs

```diff
@@ -181,14 +199,39 @@ impl Actor {
         Ok(WithdrawBalanceReturn { amount_withdrawn: amount_extracted })
     }
 
+    /// Returns the escrow balance and locked amount for an address.
+    fn get_balance(
+        rt: &mut impl Runtime,
+        account: Address,
+    ) -> Result<GetBalanceReturn, ActorError> {
+        rt.validate_immediate_caller_accept_any()?;
+        let nominal = rt.resolve_address(&account).ok_or_else(|| {
+            actor_error!(illegal_argument, "failed to resolve address {}", account)
+        })?;
+        let account = Address::new_id(nominal);
+
+        let store = rt.store();
+        let st: State = rt.state()?;
+        let balances = BalanceTable::from_root(store, &st.escrow_table)
+            .context_code(ExitCode::USR_ILLEGAL_STATE, "failed to load escrow table")?;
+        let locks = BalanceTable::from_root(store, &st.locked_table)
+            .context_code(ExitCode::USR_ILLEGAL_STATE, "failed to load locked table")?;
+        let balance = balances
+            .get(&account)
+            .context_code(ExitCode::USR_ILLEGAL_STATE, "failed to get escrow balance")?;
+        let locked = locks
+            .get(&account)
+            .context_code(ExitCode::USR_ILLEGAL_STATE, "failed to get locked balance")?;
+
+        Ok(GetBalanceReturn { balance, locked })
+    }
+
     /// Publish a new set of storage deals (not yet included in a sector).
     fn publish_storage_deals(
         rt: &mut impl Runtime,
         params: PublishStorageDealsParams,
     ) -> Result<PublishStorageDealsReturn, ActorError> {
-        // Deal message must have a From field identical to the provider of all the deals.
-        // This allows us to retain and verify only the client's signature in each deal proposal itself.
-        rt.validate_immediate_caller_type(CALLER_TYPES_SIGNABLE.iter())?;
+        rt.validate_immediate_caller_accept_any()?;
```

Comment:
Double checking here: the implicit assumption with exposing this is that miner control addresses will be contracts because the caller is constrained to be a miner control address.  I want to make sure we are communicating this requirement well because it might be unexpected for users who want to use the storage market.  It's possible this is too confusing and we might want to keep PSD from being exported.

Reply:
Yes, i think you're right, and the suggestion to externalize this information is well-taken. This method now _can_ be called by non-builtin actors, but those actors _must_ still be control addresses.